### PR TITLE
[ERSSUP-67731]-[]-[Upgrade test utils to 2.0]-[DMW]

### PR DESCRIPTION
**Pull request by nhsd-david-wass**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: tests/integration/test_end_to_end.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 _VALID_NHSD_TARGET_IDENTIFIER_BASE_64 = base64.b64encode(bytes(json.dumps(_VALID_NHSD_TARGET_IDENTIFIER), 'utf-8'))
 
 _EXPECTED_CORRELATION_ID = "123123-123123-123123-123123"
-_MATCH_PATIENT_NHS_NUMBER = "9912003888"
+_MATCH_PATIENT_NHS_NUMBER = "9900000285"
```
Comment (nhsd-jack-wainwright) [inline tests/integration/test_end_to_end.py:21]: Could this be swapped to utilise the NHS number fixture?

File: tests/integration/test_headers.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 _VALID_NHSD_TARGET_IDENTIFIER_BASE_64 = base64.b64encode(bytes(json.dumps(_VALID_NHSD_TARGET_IDENTIFIER), 'utf-8'))
 
 _EXPECTED_CORRELATION_ID = "123123-123123-123123-123123"
-_MATCH_PATIENT_NHS_NUMBER = "9912003888"
+_MATCH_PATIENT_NHS_NUMBER = "9900000285"
```
Comment (nhsd-jack-wainwright) [inline tests/integration/test_headers.py:27]: I think this could also use the NHS number fixture

**Review by nhsd-jack-wainwright — COMMENTED** [review]
Looks good to me, just a few minor comments.

File: sandbox/package-lock.json
[the remark this answers is not in the document]
Comment (nhsd-david-wass) [reply sandbox/package-lock.json:1]: yes - its been needed to let this happen for a while

File: tests/api_tests.py [hunk 3]
```diff
@@ -1,82 +1,38 @@
...
-    """
-
-    is_deployed = partial(_is_deployed_and_connected, api_test_config=api_test_config)
-
-    await poll_until(
-        make_request=lambda: api_client.get('_status', headers={'apikey': env.status_endpoint_api_key()}),
-        until=is_deployed,
-        timeout=120
-    )
+class TestStatusEndpoints:
```
[the remark this answers is not in the document]
Comment (nhsd-david-wass) [reply tests/api_tests.py:8]: the whole class is marked with @pytest.mark.smoketest

File: tests/integration/test_end_to_end.py (hunk 1 above)
Comment (nhsd-david-wass) [reply tests/integration/test_end_to_end.py:21]: Its used in a '@pytest.mark.parametrize' which doesn't/is very hard to make use fixtures - I expect that is why it was specified as a global here

File: tests/api_tests.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nhsd-david-wass) [reply tests/api_tests.py:8]: I've checked they run in the pipelines as well

File: tests/conftest.py [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
+    AuthorizationCodeConfig,
+    AuthorizationCodeAuthenticator,
+)
+from pytest_nhsd_apim.apigee_apis import (
+    ApiProductsAPI,
+    ApigeeClient,
+    ApigeeNonProdCredentials,
+    DeveloperAppsAPI,
+)
 __JWKS_RESOURCE_URL = "https://raw.githubusercontent.com/NHSDigital/identity-service-jwks/main/jwks/internal-dev/9baed6f4-1361-4a8e-8531-1f8426e3aba8.json"
```
[the remark this answers is not in the document]
Comment (nhsd-david-wass) [reply tests/conftest.py]: Removed and replaced with fixture from pytest-nhsd-apim

File: tests/conftest.py [hunk 5]
```diff
@@ -67,76 +69,93 @@ def status_endpoint_api_key():
...
-
-        # Assign the new app to the product
-        await app.add_api_product([product.name])
+        devAppAPI = DeveloperAppsAPI(client=client)
+        app_name = f"apim-auto-{uuid4()}"
+
+        attributes = [{"name": "DisplayName", "value": app_name}]
+
+        for key, value in custom_attributes.items():
+            attributes.append({"name": key, "value": value})
```
[the remark this answers is not in the document]
Comment (nhsd-david-wass) [reply tests/conftest.py]: done

**Review by nhsd-jack-wainwright — APPROVED** [review]

**Pull request merged**